tensorflow/core/util/gpu_device_functions.h

```diff
@@ -251,11 +261,20 @@ __device__ T CudaShuffleSync(unsigned mask, T value, int src_lane,
 __device__ inline double CudaShuffleSync(unsigned mask, double value,
                                          int src_lane, int width = warpSize) {
   unsigned lo, hi;
+#if GOOGLE_CUDA
   asm volatile("mov.b64 {%0,%1}, %2;" : "=r"(lo), "=r"(hi) : "d"(value));
   hi = CudaShuffleSync(mask, hi, src_lane, width);
   lo = CudaShuffleSync(mask, lo, src_lane, width);
   asm volatile("mov.b64 %0, {%1,%2};" : "=d"(value) : "r"(lo), "r"(hi));
   return value;
+#elif TENSORFLOW_USE_ROCM
+  uint64_t tmp = static_cast<uint64_t>(value);
```

Comment:
This doesn't look correct: the static_cast will do a conversion. You probably wanted to use reinterpret_cast, which is technically not legal either.

Reply:
memcpy is the only legal way of doing type punning in C++.